### PR TITLE
Brand new extent widget's calculate extent from layout map and bookmark functionality

### LINKED ISSUE #7263: Brand new extent widget's calculate extent from layout map and bookmark functionality (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#46722
Author: @nirvn
QGIS version: 3.24

**Brand new extent widget's calculate extent from layout map and bookmark functionality**

## Description

This PR adds two new functionalities to the extent widget: 
- a calculate [extent] from layout map feature; and
- a calculate [extent] from bookmark feature.

The features are available both in the compact and full mode for the widget.

Photo time:
image

image

In addition, for the compact mode, the menu tool button was updated to allow for a rapid single click on button action to set the extent widget to map canvas. This looks quite good in processing algorithm dialogs standing alongside a scale parameter widget:
image

### Commits tagged with [need-docs] or [FEATURE]
"[feature][ui][processing] Add a calculate from layout map function to the extent parameter widget" "[feature][ui][processing] Add a calculate from bookmark function to the extent parameter widget"

### PR DESCRIPTION
**Pull request by PhilipAlbrecht**
Goal: Fixes #7263

Ticket(s): #

- [ ] Backport to LTR documentation is requested

### CODE REVIEW

File: docs/user_manual/introduction/general_tools.rst [hunk 1]
```diff
@@ -2513,11 +2513,14 @@ The :guilabel:`Extent` selector widget is a convenient shortcut when you want to
...
-* Use current :guilabel:`Map canvas extent`
-* :guilabel:`Draw on canvas` a rectangle whose coordinates are then used
+* Use current :guilabel:`Map Canvas Extent`
+* :guilabel:`Draw on Canvas`: a rectangle whose coordinates are then used
+* Use current :guilabel:`Calculate from Bookmark`: uses extent of a layer saved in the
+  bookmarks
+* Use current :guilabel:`Calculate from Layout Map`: uses extent of the layout map
 * Enter or edit the coordinates as ``xmin, xmax, ymin, ymax``
 
 .. _figure_extentselector_widget:
```
Comment (DelazJ) [inline docs/user_manual/introduction/general_tools.rst:2526]: This screenshot may need some update (was probably extracted from the "save features as ..." dialog) as well as the ones at https://docs.qgis.org/testing/en/docs/user_manual/managing_data_source/create_layers.html#creating-new-layers-from-an-existing-layer (raster and vector - I didn't read the text around but wonder if the extent group is worth expanding :thinking: )

**Review by DelazJ — COMMENTED** [review]
Thanks. we might need some screenshots updated. And could you remove unrelated commits, please?

File: docs/user_manual/introduction/general_tools.rst [hunk 2]
```diff
@@ -2513,11 +2513,14 @@ The :guilabel:`Extent` selector widget is a convenient shortcut when you want to
...
-* :menuselection:`Calculate from layer -->`: uses extent of a layer loaded in the
+* :guilabel:`Current Layer Extent`: e.g. when exporting a layer
+* :menuselection:`Calculate from Layer -->`: uses extent of a layer loaded in the
   current project
-* Use current :guilabel:`Map canvas extent`
-* :guilabel:`Draw on canvas` a rectangle whose coordinates are then used
+* Use current :guilabel:`Map Canvas Extent`
+* :guilabel:`Draw on Canvas`: a rectangle whose coordinates are then used
+* Use current :guilabel:`Calculate from Bookmark`: uses extent of a layer saved in the
+  bookmarks
```
[the remark this answers is not in the document]
Comment (PhilipAlbrecht) [reply docs/user_manual/introduction/general_tools.rst:2522]: yes it's better like this

File: docs/user_manual/introduction/general_tools.rst (hunk 1 above)
Comment (PhilipAlbrecht) [reply docs/user_manual/introduction/general_tools.rst:2526]: Yes it's not necessary to expanding extent for this point. But i think its okay to do it. So i updatedt the screenshots

Also i updated the one for the extent selector

**Pull request merged**